### DISCUSSION TITLE
Can i use the CDN for free

### DISCUSSION THREAD

**Discussion by Devang-911** [fingerprintjs/fingerprintjs#1011] [Q&A]
i just want to create visitor id for device, and then with each login i want to match with the first id that was created
i know it has some limitation and can not identify the device itself i still thinks its better option rather then having no option

i have found with all of your releases you provide source code, so can i download and use the source code directly from there

Bye the way i have very small team like around 5-10 so all i have to do is keep sure thay use their laptop to login to my website and forgot to mention its an internal website so the users will be 10 max

waiting for answer

**Comment by Devang-911** [comment]
@makma Can you help a bit as i am in hurry

**Comment by makma** [comment]
Hello @Devang-911,
Your are confusing two products, the source available  FingerprintJS and proprietary SaaS Fingerprint. Please take a look at this [article](https://dev.fingerprint.com/docs/identification-vs-fingerprintjs), it should answer all your questions above.

Also, note there are some licence [limitations](https://fingerprint.com/blog/fingerprintjs-license-change/) when it comes to the latest version of FingerprintJS (apply to v4 only).

For your usecase, I'd follow this [guide](https://fingerprint.com/blog/stop-credential-stuffing/).

**Comment by Devang-911** [comment]
@makma 
i have read all the references you have provided and thank you for the answer, Now as i know FingerprintJS and proprieatry SaaS Fingerprint are different product.

The only thing i can see through it right now FingerprintJS is what i want to use as my use case is simple i just want a unique device id for each of the laptops my team will use, so i think FingerprintJS will do the trick

However still my concern is that as i am using it for 5 to 10 people max and all i need to do is just generate unique device id each time a user whats to log thereafter  i am gonna manually match that generated id with database

here my main point is i want to restrict specific laptop with specific user login

So for these kind of use can i use V4 for free? or the last MIT licence version wil be enough

**Comment by makma** [comment]
>so i think FingerprintJS will do the trick

The accuracy of FingerprintJS is quite limited (40%-60%), and depends mostly on the variety of devices you use. You have to do your research to see if it's sufficient for your use case and audience.

> So for these kind of use can i use V4 for free?

The v4 can't be used for production use cases without obtaining the license. You can use v3 according to the MIT license (therefore even for the prod use case for free).

**Comment by Devang-911** [comment]
@makma 
Alright!, Thanks man for your time

**Comment by Devang-911** [comment]
@makma 
Alright!, i will check

Thanks for the help
